### PR TITLE
Add taskstats support for benchmark module to get insight about each thread's behavior

### PR DESCRIPTION
**Pull request by kawamuray**
# Motivation

When I was investigating decaton's performance degaradation suspect regarding kafka-client upgrade (which was not the case though), I found that in addition to the profiling it's very helpful to have per-java thread activity information particularly, declay accounting of each tasks to distinguish whether the certain processing latency is due to wait such as I/O.
At that time I just used linux-dist supplied "getdelays.c" program with manual thread-task mapping but I made it much sophisticated and usable for jvm threads by https://github.com/kawamuray/jtaskstats and now attempting to integrate it into decaton's benchmark module.

This PR adds `--taskstats` and `--taskstats-bin` option to benchmark module to enable taskstats recording for the threads of JVM at the end of benchmark execution.

# Sample output
```
                                Task                                |  utime  |  stime  |    rss    |    vmem     |   read   | write  |    d:cpu    |   d:bio   | d:swap | d:reclaim 
 183 Thread 1 - main                                                | 1548000 | 1486000 | 133199517 |  8508932585 | 20465664 |   4096 |  1015070636 | 289685400 |      0 |         0 
 193 Thread 2 - Reference Handler                                   |    3000 |    1000 |    217679 |    11396523 |        0 |      0 |    18850513 |         0 |      0 |         0 
 194 Thread 3 - Finalizer                                           |    9000 |    4000 |    636425 |    37004468 |        0 |      0 |    23316256 |         0 |      0 |         0 
 195 Thread 4 - Surrogate Locker Thread (Concurrent GC)             |       0 |    1000 |    128199 |     2951324 |        0 |      0 |      709024 |         0 |      0 |         0 
 196 Thread 5 - Signal Dispatcher                                   |       0 |       0 |         0 |           0 |        0 |      0 |     2214877 |         0 |      0 |         0 
 204 Thread 9 - DecatonSubscriptionThread-decaton-benchmark         | 3306000 | 3288000 | 631455699 | 19159557294 | 14995456 | 112640 | 13270728132 | 191758100 |      0 |         0 
 205 Thread 11 - kafka-coordinator-hear...hread | decaton-benchmark |  294000 |  478000 |  78531401 |  2256268218 |   684032 |   7168 |  1081045410 |         0 |      0 |         0 
 208 Thread 12 - PartitionProcessorThre...2e0-b120-484265280e01-1-0 |   22000 |   31000 |   5568261 |   154490542 |    33792 |      0 |   314683683 |  27254300 |      0 |         0 
 209 Thread 13 - PartitionProcessorThre...2e0-b120-484265280e01-1-1 |   22000 |   31000 |   5702976 |   154960472 |        0 |      0 |   312651846 |         0 |      0 |         0 
 210 Thread 14 - PartitionProcessorThre...2e0-b120-484265280e01-1-2 |    8000 |   38000 |   5327851 |   134983796 |        0 |      0 |   220938991 |         0 |      0 |         0 
 211 Thread 15 - PartitionProcessorThre...2e0-b120-484265280e01-1-3 |   22000 |   36000 |   6476210 |   169842691 |     3072 |      0 |   375805774 |         0 |      0 |         0 
 212 Thread 16 - PartitionProcessorThre...2e0-b120-484265280e01-1-4 |   19000 |   30000 |   5415785 |   143642328 |        0 |      0 |   305531639 |         0 |      0 |         0 
 213 Thread 17 - PartitionProcessorThre...2e0-b120-484265280e01-1-5 |   18000 |   39000 |   6389132 |   167199281 |        0 |      0 |   336072260 |         0 |      0 |         0 
 214 Thread 18 - PartitionProcessorThre...2e0-b120-484265280e01-1-6 |   21000 |   33000 |   6177382 |   158346855 |     3072 |      0 |   226672490 |         0 |      0 |         0 
 215 Thread 19 - PartitionProcessorThre...2e0-b120-484265280e01-1-7 |   16000 |   37000 |   6173531 |   155745015 |        0 |      0 |   288567211 |         0 |      0 |         0 
 216 Thread 20 - PartitionProcessorThre...2e0-b120-484265280e01-1-8 |   19000 |   38000 |   6449773 |   167228828 |        0 |      0 |   306097892 |         0 |      0 |         0 
 217 Thread 21 - PartitionProcessorThre...2e0-b120-484265280e01-1-9 |   22000 |   45000 |   7360941 |   196116554 |    53248 |      0 |   314293689 |         0 |      0 |         0 
 218 Thread 22 - PartitionProcessorThre...e0-b120-484265280e01-1-10 |   27000 |   41000 |   7368663 |   198717382 |    65536 |      0 |   350077935 |         0 |      0 |         0 
 219 Thread 23 - PartitionProcessorThre...e0-b120-484265280e01-1-11 |   22000 |   30000 |   5820847 |   152612714 |        0 |      0 |   284143515 |         0 |      0 |         0 
 220 Thread 24 - PartitionProcessorThre...e0-b120-484265280e01-1-12 |   16000 |   48000 |   7049648 |   187692902 |        0 |      0 |   307241172 |         0 |      0 |         0 
 221 Thread 25 - PartitionProcessorThre...e0-b120-484265280e01-1-13 |   28000 |   36000 |   7299847 |   187952910 |        0 |      0 |   282594320 |         0 |      0 |         0 
 222 Thread 26 - PartitionProcessorThre...e0-b120-484265280e01-1-14 |   22000 |   40000 |   6821574 |   181494589 |    17408 |      0 |   257832096 |         0 |      0 |         0 
 223 Thread 27 - PartitionProcessorThre...e0-b120-484265280e01-1-15 |   25000 |   38000 |   6918281 |   184725507 |        0 |      0 |   320362396 |         0 |      0 |         0 
 224 Thread 28 - PartitionProcessorThre...e0-b120-484265280e01-1-16 |   23000 |   33000 |   6216613 |   164111718 |        0 |      0 |   412529264 |         0 |      0 |         0 
 225 Thread 29 - PartitionProcessorThre...e0-b120-484265280e01-1-17 |   25000 |   38000 |   6808144 |   184579808 |    23552 |      0 |   421818386 |         0 |      0 |         0 
 226 Thread 30 - PartitionProcessorThre...e0-b120-484265280e01-1-18 |   17000 |   25000 |   4741417 |   123258386 |        0 |      0 |   218424272 |  27443100 |      0 |         0 
```

### CODE REVIEW

File: benchmark/run-bm-docker.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (ocadaruma) [inline benchmark/run-bm-docker.sh:1]: - It'd be nice to add a instruction to how to build docker image and run `run-bm-docker.sh` in README

**Comment by kawamuray** [comment]
> one question: utime, stime seem millisecond resolution always (last digits are 000). Is that a spec?

Yes. They're in usec precision from the original source but is converted and showed in nsecs for consistency: https://github.com/kawamuray/linux-taskstats-rs/blob/master/src/model.rs#L132

File: benchmark/run-bm-docker.sh (hunk 1 above)
Comment (kawamuray) [reply benchmark/run-bm-docker.sh:1]: ok, let me add.

**Comment by ocadaruma** [comment]
Ah usec and nanosec. I see thanks

**Pull request merged**